packages/runtime/src/templates/edge-shared/utils.ts

```diff
@@ -270,6 +270,9 @@ export const buildResponse = async ({
     }
     res.headers.set('x-middleware-rewrite', relativeUrl)
 
+    request.headers.set('x-original-path', new URL(request.url, `http://n`).pathname)
```

Comment:
This is just a "one-liner" trick to get `pathname` part of url when we don't know if first argument is fully qualified url or just pathname. We do discard `host` part anyway, and pick just `.pathname` out of constructed URL object.

So this will work for both:
 - `new URL('/wat', 'http://n').pathname` -> `/wat`
 - `new URL('http://example.com/wat', 'http://n').pathname` -> `/wat`

The reason we need second arg is because `new URL('/wat')` would throw because it tries to construct fully qualified url:
`new URL('/wat')` would throw with `Uncaught TypeError: Failed to construct 'URL': Invalid URL` - we don't really care about domain/host but we need it to ensure it doesn't throw when constructing new `URL` instance.

That is also used around in Next codebase a bit and reason for me using that particular `http://n` base (tho any would do) - 
https://github.com/vercel/next.js/blob/7dfa56c7141ae2556af0d7fd837e0af686d8fa55/packages/next/src/shared/lib/router/router.ts#L1073

Reply:
Aha! Makes sense. Thanks for the trick!